invenio_rdm_records/services/services.py

```diff
@@ -405,6 +405,32 @@ def reserve_pid(self, id_, identity, pid_type, pid_client=None):
             links_tpl=self.links_item_tpl,
         )
 
+    def resolve_pid(
+        self, id_, identity, pid_type, pid_prefix=None, pid_client=None
+    ):
+        """Resolve PID to a record."""
+        # PIDS-FIXME: should be migrated to self.record_cls.pids.resolve()
+        provider = self.get_provider(pid_type, client_name=pid_client)
+
+        pid_value = id_
+        if pid_prefix:
+            pid_value = f"{pid_prefix}/{pid_value}"
+
+        pid = provider.get(pid_value=pid_value, pid_type=pid_type)
+
+        # get related record/draft
+        record = self.record_cls.get_record(pid.object_uuid)
+
+        # permissions
+        self.require_permission(identity, "manage", record=record)
+
+        return self.result_item(
```

Comment:
Can we simply return the `link` only?
Ideally, it would be great to simply fetch the PID with `PersistentIdentifer.get` and generate the link, without fetching the record... not sure it is easy.

Reply:
Discussed IRL... Not as it is right now to make it consistent.